### PR TITLE
[JENKINS-59412] No reason why `User/builds` or `View/builds` should not work with Pipeline

### PR DESCRIPTION
**Pull request by jglick**
See [JENKINS-59412](https://issues.jenkins-ci.org/browse/JENKINS-59412). #4207 was mistaken. While `Computer/builds` is truly incompatible with Pipeline for reasons explained in [JENKINS-38877](https://issues.jenkins.io/browse/JENKINS-38877), `User/builds` can be made compatible with only simple changes, and `View/builds` was already fine.

### Proposed changelog entries

* Displaying Pipeline builds among user build history, and removing incorrect warning about view build history.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]

File: core/src/main/java/hudson/model/User.java
Comment (jtnord) [inline core/src/main/java/hudson/model/User.java:713]: Code is all fine and dandy, bit the javadoc comment was wrong previously (this includes SCM change and triggers by the user). Would make a suggestion but outside of the code diff.

**Review by jtnord — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by res0nance — APPROVED** [review]

**Pull request merged**
